libs/server-sdk/src/data_sources/streaming_data_source.cpp

```diff
@@ -0,0 +1,152 @@
+#include <boost/asio/any_io_executor.hpp>
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/post.hpp>
+#include <boost/json.hpp>
+#include <boost/url.hpp>
+
+#include <utility>
+
+#include "streaming_data_source.hpp"
+
+#include <launchdarkly/network/http_requester.hpp>
+
+namespace launchdarkly::server_side::data_sources {
+
+static char const* const kCouldNotParseEndpoint =
+    "Could not parse streaming endpoint URL.";
+
+static char const* DataSourceErrorToString(launchdarkly::sse::Error error) {
+    switch (error) {
+        case sse::Error::NoContent:
+            return "server responded 204 (No Content), will not attempt to "
+                   "reconnect";
+        case sse::Error::InvalidRedirectLocation:
+            return "server responded with an invalid redirection";
+        case sse::Error::UnrecoverableClientError:
+            return "unrecoverable client-side error";
+    }
+}
+
+StreamingDataSource::StreamingDataSource(
+    config::shared::built::ServiceEndpoints const& endpoints,
+    config::shared::built::DataSourceConfig<config::shared::ServerSDK> const&
+        data_source_config,
+    config::shared::built::HttpProperties http_properties,
+    boost::asio::any_io_executor ioc,
+    IDataSourceUpdateSink& handler,
+    DataSourceStatusManager& status_manager,
+    Logger const& logger)
+    : exec_(std::move(ioc)),
+      logger_(logger),
+      status_manager_(status_manager),
+      data_source_handler_(
+          DataSourceEventHandler(handler, logger, status_manager_)),
+      http_config_(std::move(http_properties)),
+      streaming_config_(
+          std::get<config::shared::built::StreamingConfig<
+              config::shared::ServerSDK>>(data_source_config.method)),
+      streaming_endpoint_(endpoints.StreamingBaseUrl()) {}
+
+void StreamingDataSource::Start() {
+    status_manager_.SetState(DataSourceStatus::DataSourceState::kInitializing);
+
+    auto updated_url = network::AppendUrl(streaming_endpoint_,
+                                          streaming_config_.streaming_path);
+
+    // Bad URL, don't set the client. Start will then report the bad status.
+    if (!updated_url) {
+        LD_LOG(logger_, LogLevel::kError) << kCouldNotParseEndpoint;
+        status_manager_.SetState(
+            DataSourceStatus::DataSourceState::kOff,
+            DataSourceStatus::ErrorInfo::ErrorKind::kNetworkError,
+            kCouldNotParseEndpoint);
+        return;
+    }
+
+    auto uri_components = boost::urls::parse_uri(*updated_url);
+
+    // Unlikely that it could be parsed earlier, and it cannot be parsed now.
+    if (!uri_components) {
+        LD_LOG(logger_, LogLevel::kError) << kCouldNotParseEndpoint;
+        status_manager_.SetState(
+            DataSourceStatus::DataSourceState::kOff,
+            DataSourceStatus::ErrorInfo::ErrorKind::kNetworkError,
+            kCouldNotParseEndpoint);
+        return;
+    }
+
+    // TODO: Initial reconnect delay. sc-204393
```

Comment:
Likely this should still be addressed as an independent change, but maybe this ticket should be pulled into the epic.
I think that the SSE client currently has a hardcoded value and it just needs to expose a value for building instead, and then we can use that value.